### PR TITLE
feat: add CreateCase task for Cases incident management

### PR DESCRIPTION
**Pull request by brian-mulier-p**
- Adds `io.kestra.plugin.kestra.ee.cases.CreateCase`, a task that opens a Case (or attaches to a matching open one, when `linkMatchingExecutions: true`) via the Cases API — typically used under `errors:` so an incident is opened whenever a flow fails
- SLA targets (`acknowledgement`, `resolution`) are both counted from case creation
- Pinned to a local `kestra-api-client` SNAPSHOT until the client-sdk Cases bindings PR (kestra-io/client-sdk#370) merges and publishes a real version

Companion to kestra-ee PR #9383 (Cases feature) and client-sdk PR #370.

### CODE REVIEW

File: src/main/java/io/kestra/plugin/kestra/ee/cases/CreateCase.java [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+    private Property<String> caseDescription;
+
+    @Schema(title = "Case severity", description = "Defaults to `MEDIUM` if not set.")
+    @PluginProperty(group = "main")
+    private Property<CaseSeverity> severity;
+
+    @Schema(title = "Initial case status")
+    @Builder.Default
+    @PluginProperty(group = "main")
+    private Property<CaseStatus> status = Property.ofValue(CaseStatus.OPEN);
```
Comment (fdelbrayelle) [inline src/main/java/io/kestra/plugin/kestra/ee/cases/CreateCase.java:110]: 🟠 [Guidelines] `status` missing default documentation
Problem: `@Schema(title = "Initial case status")` has no `description`, so the enforced client-side default (`CaseStatus.OPEN`, via `@Builder.Default`) isn't documented anywhere — violates "Defaults must be documented in @Schema(description = ...)".
Fix: Add a description, e.g. `description = "Defaults to OPEN."`, matching the pattern used for `linkMatchingExecutions` below.

File: src/main/java/io/kestra/plugin/kestra/ee/cases/CreateCase.java [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+        )
+    }
+)
+public class CreateCase extends AbstractKestraTask implements RunnableTask<CreateCase.Output> {
+
+    @Schema(
+        title = "Namespace the case belongs to",
+        description = "Defaults to the flow's namespace."
+    )
+    @PluginProperty(group = "main")
```
Comment (fdelbrayelle) [inline src/main/java/io/kestra/plugin/kestra/ee/cases/CreateCase.java]: 🟠 [Guidelines] Every property grouped under "main"
Problem: All 14 top-level properties (namespace, title, caseDescription, severity, status, sla, linkMatchingExecutions, caseId, executionId, assignees, watchers, assetIds, labels, actions) use `@PluginProperty(group = "main")`. Per the standard group table, `main` is meant for required/primary-intent properties; dumping everything there defeats the UI grouping and reviewer expectation.
Fix: Split into the standard groups, e.g. `caseId`/`executionId`/`linkMatchingExecutions` → `reliability` (dedup/attach logic), `assignees`/`watchers`/`labels`/`actions`/`assetIds` → `processing`, keep `namespace`/`title`/`caseDescription`/`severity`/`status`/`sla` under `main`.

File: src/main/java/io/kestra/plugin/kestra/ee/cases/CreateCase.java [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+        }
+        return null;
+    }
+
+    @Builder
+    @AllArgsConstructor
+    @NoArgsConstructor
+    @Getter
+    @ToString
+    public static class SubjectsProperty {
```
Comment (fdelbrayelle) [inline src/main/java/io/kestra/plugin/kestra/ee/cases/CreateCase.java:319]: 🟠 [Guidelines] Nested property classes missing @PluginProperty group
Problem: `SubjectsProperty.users`/`.groups`, `SlaProperty.acknowledgement`/`.resolution` (line 320+), and `CaseActionProperty.label`/`.namespace`/`.flowId` (line 333+) carry only `@Schema`, no `@PluginProperty` at all — no group is declared for any of these fields, violating the mandatory "every property must declare a group" rule.
Fix: Add `@PluginProperty(group = "...")` to each nested field, mirroring the enclosing `sla`/`assignees`/`watchers`/`actions` fields' groups.

File: src/main/java/io/kestra/plugin/kestra/ee/cases/CreateCase.java [hunk 4]
```diff
@@ -0,0 +1,353 @@
...
+    @Schema(
+        title = "Attach to an existing case instead of creating one",
+        description = "When set, the triggering execution is attached to this case id instead of creating a new case. Used by the auto-generated \"keep attaching matching executions\" system flow."
+    )
+    @PluginProperty(group = "main")
+    private Property<String> caseId;
+
+    @Schema(
+        title = "The execution id to link",
+        description = "Defaults to the current task run's own triggering execution. Set this to `{{ trigger.executionId }}` when this task runs inside a Flow-trigger-driven system flow (e.g. the auto-generated \"keep attaching matching executions\" flow), so the upstream triggering execution is linked instead of the system flow's own execution."
```
Comment (fdelbrayelle) [inline src/main/java/io/kestra/plugin/kestra/ee/cases/CreateCase.java:133]: 🟠 [Guidelines/UX] `executionId` reinvents an existing core API
Problem: The described primary use case for this field — "this task runs inside a Flow-trigger-driven system flow, where the current execution is the system flow's own rather than the upstream execution" — is exactly what `RunContext#getTriggerExecutionId()` already answers. Requiring users to manually template `{{ trigger.executionId }}` instead exposes a decision Kestra could make automatically, contradicting "remove decisions that Kestra or the SDK can make for them".
Fix: Default `rExecutionId` to `runContext.getTriggerExecutionId()` when non-null (falling back to `taskRunInfo.executionId()`), and keep the explicit `executionId` property only as an escape hatch for exotic cases — or drop it if the auto-detected value always covers the described system-flow scenario.

File: src/main/java/io/kestra/plugin/kestra/ee/cases/CreateCase.java [hunk 5]
```diff
@@ -0,0 +1,353 @@
...
+            .flowId(flowInfo.id())
+            .taskId(this.id)
+            .executionId(rExecutionId)
+            .executionState(executionState)
+            .caseId(rCaseId);
+
+        Map<String, Object> result = kestraClient(runContext).cases().createFromTask(rTenantId, request);
+
+        return Output.builder()
+            .caseId((String) result.get("caseId"))
```
Comment (fdelbrayelle) [inline src/main/java/io/kestra/plugin/kestra/ee/cases/CreateCase.java]: 🟠 [Security A10 / Guidelines] Unchecked casts on raw Cases API response
Problem: `result.get("caseId")`/`result.get("created")` are cast without checking `result` is non-null or that the keys exist with the expected type. A response-shape drift on the Cases API surfaces as an opaque `NullPointerException`/`ClassCastException` with no actionable message, instead of a clear "empty/unexpected response from Cases API" error.
Fix: Guard `result == null` (and ideally validate key presence/type) before casting, throwing a message that names the failing field, per the "Null or empty response body before typed deserialization" and "No bare .get() on possibly-missing data" guidelines.

File: src/main/java/io/kestra/plugin/kestra/namespaces/List.java [hunk 6]
```diff
@@ -115,17 +118,23 @@ public List.Output run(RunContext runContext) throws Exception {
 
         KestraClient kestraClient = kestraClient(runContext);
         java.util.List<String> allNamespaces = new ArrayList<String>();
+        java.util.List<QueryFilter> filters = ns.isEmpty() ? null : java.util.List.of(
```
Comment (fdelbrayelle) [inline src/main/java/io/kestra/plugin/kestra/namespaces/List.java:121]: 🟡 [Guidelines] No regression test for the prefix-filter fix
Problem: This replaces a broken positional `q` param with a proper `QueryFilter`-based prefix filter (per the commit message, rc1's `q` param was never bound server-side), but there is no dedicated test asserting `prefix` actually narrows results — `AbstractKestraTaskTest` only exercises this class for auth-error paths.
Fix: Add a Testcontainers-based test (mirroring `NamespacesWithFlowsTest`) that creates namespaces with distinct prefixes and asserts `prefix` filtering returns only the matching ones, so this regression can't silently reappear.

**Review by fdelbrayelle — CHANGES_REQUESTED** [review]
## Kestra Plugin Code Review

### Business Requirements — partial
No linked GitHub issue (only companion PRs kestra-ee#9383 and client-sdk#370 are referenced), so this is checked against the PR's own description. The core ask (open/attach a Case, SLA targets from creation, `linkMatchingExecutions`, `caseId`/`executionId` override) is implemented and matches the description. Two mandatory documentation deliverables are missing (see Guidelines), and the PR description's "pinned to a local SNAPSHOT" claim is stale — the shipped diff actually pins to the real, already-published `kestra-api-client:2.0.0-rc2` (verified on Maven Central, contains the `Cases` model classes) — please correct the description before merge so reviewers aren't misled about build reproducibility.

### Kestra Guidelines — 6 findings
- 🟠 Nested property classes (`SubjectsProperty`, `SlaProperty`, `CaseActionProperty`) have zero `@PluginProperty` on any field — no `group` declared at all, contradicting the mandatory "every property must declare a group" rule.
- 🟠 Every one of the 14 top-level properties is declared `@PluginProperty(group = "main")` — dumping everything into `main` defeats the purpose of the group table (e.g. `caseId`/`executionId`/`linkMatchingExecutions` read as "reliability", `assignees`/`watchers`/`labels`/`actions` read as "processing").
- 🟠 `status` has no `@Schema(description = ...)` at all, so its documented default (`OPEN`) is undocumented, unlike every sibling defaulted property in the file.
- 🟠 `executionId` reimplements `RunContext#getTriggerExecutionId()` (an existing typed core accessor for exactly "the execution that triggered the current one via a Flow trigger") by asking users to manually template `{{ trigger.executionId }}`. This is precisely the system-flow scenario the PR describes as the primary motivation for the field — the task should default to `getTriggerExecutionId()` (falling back to `taskRunInfo.executionId()`) instead of exposing a decision Kestra can already make for the user.
- 🟠 `src/main/resources/doc/io.kestra.plugin.kestra.md` was not updated — the new EE `cases.CreateCase` task is entirely absent from the "Enterprise Edition tasks" section, violating the mandatory plugin how-to doc rule.
- 🟠 `AGENTS.md` was not updated with the new `io.kestra.plugin.kestra.ee.cases.CreateCase` class / `cases` subpackage, violating the mandatory AGENTS.md maintenance rule.

### Security (OWASP Top 10:2025 + KPS) — 1 finding
- 🟠 A10:2025 Mishandling of Exceptional Conditions — `CreateCase.run()` casts `result.get("caseId")`/`result.get("created")` from the raw `Map<String, Object>` returned by `cases().createFromTask()` with no null-check on `result` and no validation of key presence/type. A response shape drift (renamed/omitted key, unexpected type) surfaces as an opaque `NullPointerException`/`ClassCastException` instead of an actionable error. No other security issues found: no secret leakage in `Output`, no new insecure defaults, dependency bump is a pinned, published version (not a range/SNAPSHOT despite the PR description), no injection/deserialization concerns.

### Performance — 0 findings
✅ No issues found. `CreateCase` allocations are bounded by user-declared config, no loops over unbounded/paginated data. Bonus: `KestraTestDataUtils`'s `Integer.MAX_VALUE` → `MAX_PAGE_SIZE = 1000` change is a good defensive fix against requesting unbounded page sizes from the search API.

### Additional non-blocking notes
- 🟡 `namespaces.List`'s prefix-filter fix (broken positional `q` param → `QueryFilter`) has no dedicated regression test — `AbstractKestraTaskTest` only exercises this class for auth-error paths, and there's no `NamespacesListTest` verifying `prefix` actually narrows results against a real container.
- 🟡 `CreateCaseTest` only exercises `title`/`severity`/`caseId`/`executionId`/`linkMatchingExecutions`; the more fragile nested-serialization surface (`labels`, `actions`, `assignees`, `watchers`, `assetIds`, `sla`) has no end-to-end happy-path coverage against the real Cases API.
- 🟡 `.github/setup-unit.sh` still pre-pulls `ghcr.io/kestra-io/kestra-ee:latest`, but `AbstractKestraEeContainerTest` now starts a version-pinned tag (`v${kestraVersion}-no-plugins`) — the pre-pull is now silently dead weight and should be aligned to the same tag.
- 🟢 `severity`'s doc claims a `MEDIUM` default but, unlike `status`/`linkMatchingExecutions`, no client-side default is enforced (`orElse(null)`) — relies entirely on server behavior; consider aligning the default-handling strategy across sibling properties for consistency/testability.

**Verdict:** REQUEST CHANGES

File: src/main/java/io/kestra/plugin/kestra/ee/cases/CreateCase.java (hunk 4 above)
Comment (brian-mulier-p) [reply src/main/java/io/kestra/plugin/kestra/ee/cases/CreateCase.java:133]: getTriggerExecutionId() is just the new execution's own freshly-generated id (only set during trigger evaluation, not inside a task's context) — it's not the upstream execution, that's what {{ trigger.executionId }} already gives us, so keeping it as is.

File: src/main/java/io/kestra/plugin/kestra/ee/cases/CreateCase.java [hunk 7]
```diff
@@ -0,0 +1,372 @@
...
+    @Schema(
+        title = "Attach to an existing case instead of creating one",
+        description = "When set, the triggering execution is attached to this case id instead of creating a new case. Used by the auto-generated \"keep attaching matching executions\" system flow."
+    )
+    @PluginProperty(group = "source")
+    private Property<String> caseId;
+
+    @Schema(
+        title = "The execution id to link",
+        description = "Defaults to the current task run's own triggering execution. Set this to `{{ trigger.executionId }}` when this task runs inside a Flow-trigger-driven system flow (e.g. the auto-generated \"keep attaching matching executions\" flow), so the upstream triggering execution is linked instead of the system flow's own execution."
```
Comment (fdelbrayelle) [inline src/main/java/io/kestra/plugin/kestra/ee/cases/CreateCase.java:133]: 🟠 [Guidelines/UX] Still open from prior review (5051117320)
Problem: `executionId` still requires users to manually template `{{ trigger.executionId }}` for the Flow-trigger-driven system-flow scenario, instead of the task defaulting to `RunContext#getTriggerExecutionId()` (an existing core accessor for exactly this case). This commit addressed the other 5 findings but left this one untouched.
Fix: Default `rExecutionId` to `runContext.getTriggerExecutionId()` when non-null, falling back to `taskRunInfo.executionId()`; keep the explicit property only as an override for edge cases.

File: src/test/java/io/kestra/plugin/kestra/namespaces/ListTest.java [hunk 8]
```diff
@@ -6,66 +6,40 @@
...
+public class ListTest extends AbstractKestraOssContainerTest {
 
     @Inject
     protected RunContextFactory runContextFactory;
 
-    protected static final String NAMESPACE = "kestra.tests.namespaces.list";
-
     @Test
-    public void shouldListNamespaces() throws Exception {
+    public void shouldFilterNamespacesByPrefix() throws Exception {
```
Comment (fdelbrayelle) [inline src/test/java/io/kestra/plugin/kestra/namespaces/ListTest.java:28]: 🟠 [Guidelines] New: pagination test coverage deleted, not replaced
Problem: This rewrite removes `shouldListNamespacesWithPagination` (the only test exercising `namespaces.List`'s explicit `page` param and its multi-page iteration loop) while adding `shouldFilterNamespacesByPrefix`. Pagination behavior — unrelated to the prefix-filter bug this commit fixes — is now completely untested.
Fix: Keep a pagination test alongside the new prefix-filter test (adapt it to use a randomized/isolated namespace prefix like this new test does, to avoid the cross-test pollution that likely masked the original bug).

**Review by fdelbrayelle — CHANGES_REQUESTED** [review]
## Kestra Plugin Code Review — Re-review of fix commit `6019126` (addresses review #5051117320)

### Prior findings — status
- ✅ **Resolved** — Nested property classes (`SubjectsProperty`, `SlaProperty`, `CaseActionProperty`) now carry `@PluginProperty(group = ...)` on every field.
- ✅ **Resolved** — Properties redistributed across `main`/`source`/`reliability`/`advanced` instead of all being `main`. Minor nit only: `namespace`/`caseId`/`executionId` land in `source` even though `namespace` is really the case's target location (arguably `destination`) — not blocking.
- ✅ **Resolved** — `status` now documents its `OPEN` default via `@Schema(description = "Defaults to `OPEN`.")`.
- ✅ **Resolved** — `createFromTask` response is now guarded: `result == null` check plus `instanceof String`/`instanceof Boolean` pattern-matched guards with actionable `IllegalStateException` messages before use, replacing the blind casts.
- ✅ **Resolved** — `io.kestra.plugin.kestra.md` gained an `ee/cases.CreateCase` entry under Enterprise Edition tasks; `AGENTS.md` gained the `io.kestra.plugin.kestra.ee.cases.CreateCase` bullet in Key Plugin Classes.
- ✅ **Resolved** — `namespaces.List`'s prefix filter now has a dedicated regression test (`ListTest#shouldFilterNamespacesByPrefix`) using per-run randomized prefixes to assert a matching namespace is returned and a non-matching one is excluded.
- ❌ **Still open** — `executionId` (CreateCase.java:131-136, resolution logic at :194-201) is unchanged: it still requires users to manually template `{{ trigger.executionId }}` for the system-flow scenario instead of defaulting to `RunContext#getTriggerExecutionId()`. Not addressed by this commit.

### New issue introduced by this fix commit
- 🟠 [Guidelines] **Test coverage regression in `ListTest.java`** — Problem: the rewrite that adds `shouldFilterNamespacesByPrefix` entirely deletes `shouldListNamespacesWithPagination` (the only test exercising `namespaces.List`'s explicit `page` parameter and its multi-page `do/while` iteration loop) without any replacement. `namespaces.List`'s pagination branches are now completely untested, even though that logic is unrelated to and unaffected by the prefix-filter fix this commit set out to make. Fix: restore a pagination test (adapted to the new isolated/randomized-prefix style to avoid the cross-test pollution that likely masked the original prefix bug) alongside the new prefix-filter test, rather than replacing one with the other.

### Kestra Guidelines — 1 open (carried over), 1 new
- 🟠 `executionId` still duplicates `RunContext#getTriggerExecutionId()` (see original review comment) — the primary system-flow use case this property exists for could be auto-detected instead of requiring users to know and template `{{ trigger.executionId }}`.
- 🟠 Pagination test coverage regression in `ListTest.java` (see above).

### Security — ✅ No issues found
Unchecked-cast finding from the previous round is fully resolved with explicit null/type guards and actionable error messages.

### Performance — ✅ No issues found
No new allocation, threading, or unbounded-growth concerns in this commit.

**Verdict:** REQUEST CHANGES

**Review by fdelbrayelle — APPROVED** [review]
